pkg/login/social/connectors/common.go

```diff
@@ -118,9 +118,19 @@ func createOAuthConfig(info *social.OAuthInfo, cfg *setting.Cfg, defaultName str
 		authStyle = oauth2.AuthStyleAutoDetect
 	}
 
+	var clientSecret string
+	switch info.ClientAuthentication {
+	case "client_secret_post":
+		clientSecret = info.ClientSecret
+	case "managed_identity":
+		clientSecret = ""
```

Comment:
@kalleep here is the added logic for not setting `ClientSecret` if `client_authentication` is set to `managed_identity`. I have tested this and validated that it does in fact not set it if we use `managed_identity`, and if `client_authentication` isn't set at all it just defaults to using `client_secret_post` (hence default in the switch statement).

If all is good, then we may move onto the front end in which I am a little less informed on so I will need some guidance as to where to make the changes! Thanks again for all of your help at getting this solution implemented.

Reply:
Great. We also need to update sample.ini with the new setting

Reply:
For the rest of the work I will bring it up to my team how we can split it up because it can be a lot of work.

This includes:
* Updates / changes to front-end and sso settings api to allow this configuration
* Updates to [terraform](https://github.com/grafana/terraform-provider-grafana)
* Updates to documentation

If you are up to do the changes we can contain it within the same pr. Otherwise we can merge once I have tested it out and done a proper review. Then we can do the other required steps in followup pr:s

Reply:
@kalleep I just updated the sample.ini file! As far as remaining work goes, I would be happy to work on the first bullet point, so that we can bring this experience back to our team in case we have any other feature requests of any kind. I will need guidance on how to do this so that it can be done efficiently. If you need any help with testing I would be happy to help as well (since these changes are Azure specific at the moment)

Reply:
Hello @kalleep! I just pushed the frontend changes to this PR so please feel free to look over them and let me know if you have any feedback for changing anything. Provided is a screenshot of the changes I made for AzureAD:
<img width="487" alt="Screenshot 2024-11-27 092157" src="https://github.com/user-attachments/assets/188b98b5-de41-4e70-bf54-1ec36cc6e2f0">

Note: I only added the client authentication field to the AzureAD oauth page but if you would like to add it to every provider, I am able to do so.

Reply:
Hi @kalleep! Just checking in to see if there is any update on progress for the last two bullet points. I am happy to contribute on either remaining bullet point with your guidance on what needs changed!

Reply:
Nice, @mgyongyosi could you take a look at the changes done for sso settings. Not sure all the places we need to update

Reply:
@kalleep I am also making the documentation changes to `docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md` for the new auth method for AzureAD. These changes will be included in the PR.

Reply:
@kalleep Just pushed the changes for the Azure AD documentation!